### ISSUE TITLE
feature: Gitlab Integration

### ISSUE THREAD

**Issue by that-ambuj** [hcengineering/platform#6653]
### Description of the issue
We use a self-hosted Gitlab and Sentry instance, is it on the roadmap to integrate those into huly? 

### Your environment
N/A

### Steps to reproduce
N/A

### Expected behaviour
N/A

### Actual behaviour
N/A

### Possible solutions
(Not obligatory)
N/A

**Comment by Ahabibullah** [comment]
I really wish they work on GitLab integration

**Comment by ztripez** [comment]
Same here, we using selfhosted gitlab

**Comment by anibal-aguila** [comment]
Same here, we using selfhosted gitlab

**Comment by href** [comment]
Me too!

**Comment by BunnyMan1** [comment]
I'm open to working on it and submit a PR for this.

Can the core devs give a rough guidance on how to go about implementing this?

**Comment by Uranbold** [comment]
> I'm open to working on it and submit a PR for this.
> 
> Can the core devs give a rough guidance on how to go about implementing this?

Sounds good! Any Documentation out there?

**Comment by that-ambuj** [comment]
@Uranbold I don't see a CONTRIBUTING.md in the repo. The developer docs might be on their website or one of the maintainers might help you out in that regard. 

Besides, this issue needs triaging and approval before anyone makes a PR.

**Comment by mmuenker** [comment]
This would be very nice. And I would also be open to work on a PR.

**Comment by BunnyMan1** [comment]
@haiodo @BykhovDenis 

Please provide your inputs - if we can contribute towards this feature / integration.

**Comment by Heknon** [comment]
+1

This would be really nice

**Comment by McLovinIt101** [comment]
Would be amazing. +1. Open to help on a PR if needed

**Comment by yecine06** [comment]
we are using gitlab selfhosted and it would be great to have this

**Comment by zwkevin** [comment]
+1 this would be great. We are using Gitlab Self Hosted too!

**Comment by alighaemia** [comment]
We now use self-hosted Gitlab, and this would be fantastic.

**Comment by HilkopterBob** [comment]
What do you think about it @BykhovDenis?

**Comment by haiodo** [comment]
Hi, [services/github](https://github.com/hcengineering/platform/tree/develop/services/github) could be used a source point of how integration could work, we right now focus on more system core tasks.

Github integration is working but we plan to rework it to be more clear and useful to be en example of how integration could be implemented. But it could be used as guidance.

Also we have some examples from API point of view: https://github.com/hcengineering/huly-examples/tree/main/platform-api

**Comment by that-ambuj** [comment]
Thanks for the insight @haiodo. So, my question is that can we start working on this? or should we wait for the rework to happen? as the latter option will probably require less refactoring for you guys.

**Comment by BDVGitHub** [comment]
Anny news on this?

**Comment by ART7182** [comment]
Yes! Of course! I'm really looking forward to the Gitlab integration. We using selfhosted gitlab

**Comment by ART7182** [comment]
Is there an approximate time frame?

**Comment by stefanwuthrich** [comment]
up vote

**Comment by daudln** [comment]
Eyes on this

**Comment by outbackdingo** [comment]
following

**Comment by jannis6023** [comment]
Please integrale Gitlab. The missing is currently our dealbreaker!

**Comment by that-ambuj** [comment]
Hey guys, please don't comment things like "me too" that don't add anything to this conversation. Please simply react to the top of the issue and/or subscribe to this issue to receive updates.

I'm sure the maintainers would appreciate not having to sift through so many 'interested' comments to find the core of this issue.

Let's not crowd this thread and keep it clean 🙏🙏

**Comment by steveepreston** [comment]
`Plane` added Gitlab Integration, but it's behind a Paywall

**Comment by davidmach14** [comment]
+1 This is the only feature missing to make us switch from Trello to Huly.

**Comment by Setrias** [comment]
It's been half a year, is there any progress on this ? I'm using Plane for now but I'd like to switch after GitLab integration is present.

**Comment by that-ambuj** [comment]
> It's been half a year, is there any progress on this ? I'm using Plane for now but I'd like to switch after GitLab integration is present.

This kind of entitlement isn't helping anyone dude. Either fork it and do it yourself or shut up.

**Comment by Setrias** [comment]
> > It's been half a year, is there any progress on this ? I'm using Plane for now but I'd like to switch after GitLab integration is present.
> 
> This kind of entitlement isn't helping anyone dude. Either fork it and do it yourself or shut up.

I just asked about the situation, if it moved somewhere ¯\_(ツ)_/¯.

**Comment by wafelmolt** [comment]
Hello. What's up on this subject ? It is a must have for many of us. Could the maintainers give us some overview on the roadmap status for this feature ? Thx 😃

**Comment by coolbirdzik** [comment]
Waiting

**Comment by marab2** [comment]
+1 this is needed

**Comment by Romain-GUILLEMOT** [comment]
+1 this is needed

**Comment by Nikita-Ma** [comment]
+1 needed

**Comment by wafelmolt** [comment]
Seeing many answers but no reaction from hcengineering nor Huly notif for mirror in their instance (+ the fact that the issue is very old now), I guess this issue is only present on Github side. Should we recreate the issue for it to be picked up by their integration ?

**Comment by steveepreston** [comment]
@wafelmolt
Not bad idea, create a new to move on top of list

**Comment by that-ambuj** [comment]
Yes, please feel free to close this issue. It doesn't have a lot of meaningful comments.

**Comment by ArtyomSavchenko** [comment]
Hi all
I apologize for the fact that no one from the Huly team responded for a long time. Personally, I couldn't see this task because it was created a long time ago.

At the moment we don't have enough free people to do this right now, but I think adding Gitlab integration would be useful for the product.
We will be happy to help if anyone wants to contribute. But this may not be easy and will require a lot of work.

I added a discussion for the Gitlab integration so it doesn't get lost: https://github.com/hcengineering/platform/discussions/10253

**Comment by outbackdingo** [comment]
Calling for testers — https://github.com/optim-enterprises-bv/huly-gitlab

Community Phase 1 (Issues two-way sync) + Phase 2 (Merge Requests + pipeline status) implementation of a Huly ↔ GitLab integration pod. Out-of-tree pod mirroring the existing `services/github/` shape. Reviews/approvals/diff metadata deferred to Phase 3. Feedback and bug reports welcome.

**Comment by steveepreston** [comment]
@outbackdingo Is this limited to Business/Paid version?
or can be used in Free SelfHosted?

**Comment by outbackdingo** [comment]
> [@outbackdingo](https://github.com/outbackdingo) Is this limited to Business/Paid version? or can be used in Free SelfHosted?

free for everyone, use it where ever you want as far as im concerned :)
